ui/src/app/base/components/AuthenticationFields/AuthenticationFields.test.tsx

```diff
@@ -0,0 +1,67 @@
+import { mount } from "enzyme";
+import { Formik } from "formik";
+
+import AuthenticationFields from "./AuthenticationFields";
+
+describe("AuthenticationFields", () => {
+  it("renders different fields depending on whether a certificate should be generated", () => {
+    const Proxy = ({ shouldGenerateCert }: { shouldGenerateCert: boolean }) => (
+      <Formik
+        initialValues={{ certificate: "", key: "", password: "" }}
+        onSubmit={jest.fn()}
+      >
+        <AuthenticationFields
+          onRadioChange={jest.fn()}
+          shouldGenerateCert={shouldGenerateCert}
+        />
+      </Formik>
+    );
+    const wrapper = mount(<Proxy shouldGenerateCert={true} />);
+    expect(wrapper.find("input[name='password']").exists()).toBe(true);
+    expect(wrapper.find("textarea[name='certificate']").exists()).toBe(false);
+    expect(wrapper.find("textarea[name='key']").exists()).toBe(false);
+
+    wrapper.setProps({ shouldGenerateCert: false });
```

Comment:
It doesn't look like this component needs to respond to the prop changing so it might be simpler to have two tests.

Reply:
Good point, thanks!